### PR TITLE
Added dot to paddle frontend API

### LINKED ISSUE #17866: dot

### PR DESCRIPTION
**Pull request by AmrSheta22**
Close #17866

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by AmrSheta22** [comment]
> If you are working on an open task, please **edit** the PR description to link to the issue you've created.
> 
> For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).
> 
> Thank you 🤗

Oh sorry, did I link it correctly now?

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py:665]: Please use `valid` instead

```suggestion
        available_dtypes=helpers.get_dtypes("valid"),
```

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py:666]: Is that necessary?

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py:668]: I guess if we used the default values of the strategy would be better

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hey @AmrSheta22

I've made some preliminary comments on your code. Please look at them.

While you are at it, to streamline the review process a little bit. Do you mind adding a comment and in it "add_frontend_checklist" (without double quotes nor newlines)? Our automation will convert your comment into a checklist where you can fill which points you have completed, then request a review from me.

Thank you so much for your contribution 🚀

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py
Comment (AmrSheta22) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py:665]: Gladly but the documentation in paddle specified that it uses only float, are you sure we should make that change?

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py
Comment (AmrSheta22) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py:666]: I think having it 1 dimensional is necessary for the a vector dot product.

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py
Comment (AmrSheta22) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py:668]: add_frontend_checklist

**Comment by AmrSheta22** [comment]
add_frontend_checklist

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py
Comment (AmrSheta22) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py:665]: I don't think valid is the right option here since the "dot" function only works with float in paddle, also using valid doesn't make sense to me. Can you elaborate on how this is the better option?

Comment (KareemMAX) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py:665]: `helpers.get_dtypes("valid")` should resolve valid dtypes automatically, which in turn are specified by `@with_supported_dtypes`/`@with_unsupported_dtype` decorators.

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py
Comment (KareemMAX) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py:666]: I think this is a scaler shape not a 1-dimentsional one, also paddle support 2D shapes as inputs apparently according to their [docs](https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/dot_en.html).

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hey @AmrSheta22,

Thanks for adding the checklist, please address the comment and missing points on the checklist and make sure to check them for me afterwards. Then please request a review from me.

Thanks for contributing 🚀

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py
Comment (AmrSheta22) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py:668]: Paddle has a habit of turning really large numbers into (inf) or really small numbers into (-inf) when other frameworks don't, and it causes the internal tests to fail. And that's also what others did in their tests.

Comment (KareemMAX) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py:671]: Unit test should pass on every valid case on paddle, if paddle behaves in a certain way, we should expect the frontend function to behave the same.

**Comment by KareemMAX** [comment]
ivy-gardener

File: ivy/functional/frontends/paddle/tensor/linalg.py
Comment (KareemMAX) [inline ivy/functional/frontends/paddle/tensor/linalg.py:165]: This code fragment doesn't look efficient. Can't you figure out a vectorized implementation? I'm pretty sure there is.

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hey @AmrSheta22,

I've pointed an issue with your implementation of 2D vectors dot product. In addition to my old point on testing. Please take a look at them.

Once they are resolved we can merge 🚀

Thanks 😄

File: ivy/functional/frontends/paddle/tensor/linalg.py
Comment (AmrSheta22) [reply ivy/functional/frontends/paddle/tensor/linalg.py:165]: Good point! right on it!

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py
Comment (AmrSheta22) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py:671]: Aside from the issue above (which I couldn't replicate today to be honest), the problem is that the error range of the absolute difference is around atol = 1e-06 , and the acceptable range is atol=1e-08. So I am not sure if I can get the error lower but I am going to try to make it work :) if I couldn't I will push the changes done to vectorizing the function. 

Another note: I see that most functions implemented in linalg.py file either use a range of values like I did before or they lower the acceptable error. I will try the second approach if I run out of options and leave it to your evaluation :).

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py:691]: I believe this tolerance is a bit high? Can you use safety factor instead?

**Review by KareemMAX — CHANGES_REQUESTED** [review]
I've only a small issue here 😅. Can you take a look into this? @AmrSheta22

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py
Comment (KareemMAX) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py:691]: I've actually tested your code without this line and it passes. Please remove this line then and merge with master, after which we will be ready to merge

**Review by KareemMAX — APPROVED** [review]

**Pull request merged**
